docs/tutorials/python/build-apps/credential-issuing-service.md

```diff
@@ -0,0 +1,342 @@
+
+{% code-snippet file="/_code-samples/issue-credentials/py/credential_model.py" language="py" from="    def __init__(self, d: dict):" before="@classmethod" /%}
+
+The `from_xrpl(...)` class method is an alternate constructor for the Credential class. It takes a dictionary in the XRP Ledger's native format and decodes it to the native Python formats the Credential class expects (for example, converting the `credential` field from hexadecimal to a native string). The API methods that read data from the XRP Ledger use this constructor so that their output is formatted the same way as user input in the other API methods.
+
+{% code-snippet file="/_code-samples/issue-credentials/py/credential_model.py" language="py" from="    @classmethod" before="def to_dict(self):" /%}
+
+The `to_dict(self)` method builds a dictionary representation for the Credential object, which can then be returned by the API as JSON. It converts from a `datetime` back to an ISO 8601 string and omits optional fields instead of including them with a `None` or `null` value.
+
+{% code-snippet file="/_code-samples/issue-credentials/py/credential_model.py" language="py" from="    def to_dict(self):" before="def to_xrpl(self):" /%}
+
+The `to_xrpl(self)` method returns a different class of object, `XrplCredential`, which is formatted for submitting to the XRP Ledger:
+
+{% code-snippet file="/_code-samples/issue-credentials/py/credential_model.py" language="py" from="    def to_xrpl(self):" before="class XrplCredential:" /%}
+
+The implementation of `XrplCredential` performs the necessary conversions in its constructor:
+
+{% code-snippet file="/_code-samples/issue-credentials/py/credential_model.py" language="py" from="class XrplCredential:" before="class CredentialRequest(Credential):" /%}
+
+Finally, the `CredentialRequest` class inherits from the `Credential` class but checks for an additional field, `documents`. For a realistic credential issuer, you might require the user to provide specific documents in the request body, like a photo of their government-issued ID or a cryptographically signed message from another business, which your code would check. For this tutorial, the check is only a placeholder:
+
+{% code-snippet file="/_code-samples/issue-credentials/py/credential_model.py" language="py" from="class CredentialRequest(Credential):" /%}
+
+{% admonition type="success" name="Tip" %}
+Depending on the meaning of the credential, verifying the documents might require human intervention or it might take longer than the amount of time a RESTlike API can reasonably take to respond. In those situations, you would want to save the Credential request to some kind of storage like an SQL database, and change the API to have separate methods for admins to accept or reject credential requests.
+{% /admonition %}
+
+{% raw-partial file="/docs/_snippets/common-links.md" /%}
```

Comment:
Is it worth reiterating this when you did so earlier? Maybe just move the additional info about SQL databases and creating separate methods into the earlier note.